src/encode/n_axioms.ml

```diff
@@ -0,0 +1,5029 @@
+  ~pats:[ [
+    apps T.IntMinus
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ appb B.Conj
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; apps T.Mem
+        [ Ix 1 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ] %% []
+    ; apps T.Mem
+      [ apps T.IntMinus
+        [ Ix 2 %% []
+        ; Ix 1 %% []
+        ] %% []
+      ; apps T.IntSet [] %% []
+      ] %% []
+    ] %% []
+  ) %% []
+
+let inttimes_typing () =
+  quant Forall
+  [ "x" ; "y" ] [ t_idv ; t_idv ]
+  ~pats:[ [
+    apps T.IntTimes
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ appb B.Conj
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; apps T.Mem
+        [ Ix 1 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ] %% []
+    ; apps T.Mem
+      [ apps T.IntTimes
+        [ Ix 2 %% []
+        ; Ix 1 %% []
+        ] %% []
+      ; apps T.IntSet [] %% []
+      ] %% []
+    ] %% []
+  ) %% []
+
+let intexp_typing () =
+  quant Forall
+  [ "x" ; "y" ] [ t_idv ; t_idv ]
+  ~pats:[ [
+    apps T.IntExp
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ List (And,
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; appb ~tys:[ t_idv ] B.Neq
+        [ Ix 2 %% []
+        ; apps (T.IntLit 0) [] %% []
+        ] %% []
+      ; apps T.Mem
+        [ Ix 1 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ]) %% []
+    ; apps T.Mem
+      [ apps T.IntExp
+        [ Ix 2 %% []
+        ; Ix 1 %% []
+        ] %% []
+      ; apps T.IntSet [] %% []
+      ] %% []
+    ] %% []
+  ) %% []
+
+let intquotient_typing () =
+  quant Forall
+  [ "x" ; "y" ] [ t_idv ; t_idv ]
+  ~pats:[ [
+    apps T.IntQuotient
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ List (And,
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; apps T.Mem
+        [ Ix 1 %% []
+        ; apps T.NatSet [] %% []
+        ] %% []
+      ; apps T.IntLteq
+        [ apps (T.IntLit 0) [] %% []
+        ; Ix 1 %% []
+        ] %% []
+      ]) %% []
+    ; apps T.Mem
+      [ apps T.IntQuotient
+        [ Ix 2 %% []
+        ; Ix 1 %% []
+        ] %% []
+      ; apps T.NatSet [] %% []
+      ] %% []
+    ] %% []
+  ) %% []
+
+let intremainder_typing () =
+  quant Forall
+  [ "x" ; "y" ] [ t_idv ; t_idv ]
+  ~pats:[ [
+    apps T.IntRemainder
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ List (And,
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; apps T.Mem
+        [ Ix 1 %% []
+        ; apps T.NatSet [] %% []
+        ] %% []
+      ; apps T.IntLteq
+        [ apps (T.IntLit 0) [] %% []
+        ; Ix 1 %% []
+        ] %% []
```

Comment:
Not sure I understand. Floor division and remainder are specified on the same values in TLA+ (because remainder is defined from floor division). Therefore the conditions in the code are the same.